examples/advanced_operations/add_ad_customizer.py

```diff
@@ -0,0 +1,459 @@
+    are used only for customizing a specific ad group.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID.
+        ad_group_ids: a list of ad group IDs.
+        feed_item_resource_names: a list of feed item resource name strs.
+    """
+    ad_group_service = client.get_service("AdGroupService", version="v5")
+    feed_item_target_service = client.get_service(
+        "FeedItemTargetService", version="v5"
+    )
+    # Bind each feed item to a specific ad group to make sure it will only be
+    # used to customize ads inside that ad group; using the feed item elsewhere
+    # will result in an error.
+    for i, resource_name in enumerate(feed_item_resource_names):
+        ad_group_id = ad_group_ids[0]
+
+        feed_item_target_op = client.get_type(
+            "FeedItemTargetOperation", version="v5"
+        )
+        feed_item_target = feed_item_target_op.create
+        feed_item_target.feed_item.value = resource_name
+        feed_item_target.ad_group.value = ad_group_service.ad_group_path(
+            customer_id, ad_group_id
+        )
+
+        try:
+            response = feed_item_target_service.mutate_feed_item_targets(
+                customer_id, [feed_item_target_op]
+            )
+            print(
+                "Added feed item target with resource name "
+                f"{response.results[0].resource_name}"
+            )
+        except GoogleAdsException as ex:
+            _handle_google_ads_exception(ex)
+
+
+def create_ads_with_customizations(
+    client, customer_id, ad_group_ids, feed_name
+):
+    """Creates expanded text ads that use the ad customizer feed.
+
+    The expanded text ads use the ad customizer feed to populate the
+    placeholders.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID.
+        ad_group_ids: a list of ad group IDs.
+        feed_name: the name of the feed to create.
+    """
+    ad_group_service = client.get_service("AdGroupService", version="v5")
+    ad_group_ad_service = client.get_service("AdGroupAdService", version="v5")
+    ad_group_ad_operations = []
+
+    for ad_group_id in ad_group_ids:
+        ad_group_ad_operation = client.get_type(
+            "AdGroupAdOperation", version="v5"
+        )
+        ad_group_ad = ad_group_ad_operation.create
+        ad_group_ad.ad_group = ad_group_service.ad_group_path(
+            customer_id, ad_group_id
+        )
+        ad_group_ad.ad.final_urls.append("http://www.example.com")
+        ad_group_ad.ad.expanded_text_ad.headline_part1 = (
+            f"Luxury cruise to {{={feed_name}.Name}}"
+        )
+        ad_group_ad.ad.expanded_text_ad.headline_part2 = (
+            f"Only {{={feed_name}.Price}}"
+        )
+        ad_group_ad.ad.expanded_text_ad.description = (
+            f"Offer ends in {{=countdown({feed_name}.Date)}}!"
```

Comment:
Maybe add a link to https://support.google.com/google-ads/answer/6193743?hl=en for an explanation of countdown ad customizers?

Reply:
Done